From 62b9a903b3635a171a73e69adc59843f5825b409 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Wed, 11 Jul 2018 01:29:24 +0300
Subject: [PATCH] Add BTL support fixup

memory len must be multiple of 16 otherwise output is curropted
---

diff --git a/board/habanalabs/dali/btl/btl-as.py b/board/habanalabs/dali/btl/btl-as.py
index 64044a1..4c4720a 100755
--- a/board/habanalabs/dali/btl/btl-as.py
+++ b/board/habanalabs/dali/btl/btl-as.py
@@ -48,7 +48,10 @@
     addr_hi =  int(addr,16) >> 16       # 16..43
     addr_lo =  int(addr,16) & 0xffff    # 0..15
     __opcode_addr_hi = struct.pack('>I',(_opcode << 28) + addr_hi)
-    blknum =  int(math.ceil(os.path.getsize(in_file.strip()) / 16.0))
+    length =  os.path.getsize(in_file.strip())
+    blknum =  int(math.ceil(length / 16.0))
+    # Check memory length is multiple of 16
+    assert((blknum * 16.0)==length)
     # Check BTL 64KB total limit (128b [per block] x 4K blocks)
     assert(blknum<0x1000)
     __addr_lo_blknum = struct.pack('>I', (addr_lo << 16) + blknum)
